src/actions/mod.rs

```diff
@@ -177,7 +177,7 @@ pub struct InitActionContext {
 
     // directly opened files
     pub direct_opens: Arc<Mutex<HashSet<CanonPath>>>,
-
+    pub wait_for_pending_results: Arc<AtomicBool>,
```

Comment:
From local testing, this _does_ seem to work. But I am uncertain why the linter in particular needs this consideration?

Removing all usage of this flag doesnt seem to make a difference either for or against. Perhaps the reordering on line 338 somehow fixed some ordering issue?

Do you have a sample use-case where the progress ends early without this flag, but works with it?

Reply:
This was actually one of the first things we addressed, because that was the original behavior. The client was shutting down before receiving the style diagnostics.

I'll remove the flag and try it again locally after the ordering change sending the messages.

Reply:
Confirmed, after testing on my end it does seem like this flag no longer has effect after resolving the race condition.
I'm removing it.